extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -956,6 +968,27 @@ private void enhanceEntities(final JpaEntitiesBuildItem domainObjects,
         return modelClassesPerPersistenceUnits;
     }
 
+    private static Set<String> getRelatedModelClassNames(IndexView index, Set<String> knownModelClassNames,
+            String modelClassName) {
+        Set<String> relatedModelClassNames = new HashSet<>();
+        ClassInfo modelClassInfo = index.getClassByName(DotName.createSimple(modelClassName));
+
+        // for now we only deal with entities and mapped super classes
+        if (modelClassInfo.classAnnotation(JPA_ENTITY) == null &&
+                modelClassInfo.classAnnotation(MAPPED_SUPERCLASS) == null) {
+            return Collections.emptySet();
+        }
+
+        while (modelClassInfo != null && !modelClassInfo.name().equals(DotNames.OBJECT)) {
+            if (knownModelClassNames.contains(modelClassName)) {
+                relatedModelClassNames.add(modelClassInfo.name().toString());
+            }
```

Comment:
It would probably be better to move this code just below the line below, otherwise we consider the current class too. It's not really a problem but it's less correct.

@machi1990 I'll wait to see if you have other comments before fixing this.

Reply:
Just pushed the fix to have CI running this evening.

For me it looks ready to go, waiting for your blessing.

Reply:
I was just reviewing this part. The initial version would add the model to the `relatedModelClassNames` set too. Thanks for taking care of that.